common/viewer.cpp

```diff
@@ -1345,49 +1345,61 @@ namespace rs2
             size_t index = 0;
             while (ppf.resulting_queue.poll_for_frame(&f) && ++index < ppf.resulting_queue_max_size)
             {
-                if (streams.find(f.get_profile().unique_id()) != streams.end() ||
-                    streams.find(streams_origin[f.get_profile().unique_id()]) != streams.end())
-                    last_frames[f.get_profile().unique_id()] = f;
+                // Open the frameset and save the frames on last_frames 
+                // if one of the streams is missing we will use the last frame arrived
+                if (f.is<rs2::frameset>())
+                {
+                    for (auto frame : f.as<rs2::frameset>())
+                    {
+                        if (streams.find(frame.get_profile().unique_id()) != streams.end() ||
+                            streams.find(streams_origin[frame.get_profile().unique_id()]) != streams.end())
+                            last_frames[frame.get_profile().unique_id()] = frame;
+                    }
+                }
+                else
+                {
+                    if (streams.find(f.get_profile().unique_id()) != streams.end() ||
+                        streams.find(streams_origin[f.get_profile().unique_id()]) != streams.end())
+                        last_frames[f.get_profile().unique_id()] = f;
+                }
             }
 
             for(auto&& f : last_frames)
                 not_model->output.update_dashboards(f.second);
 
-            for(auto&& frame : last_frames)
+            for( auto && frame : last_frames )
             {
                 auto f = frame.second;
-                frameset frames;
-                if ((frames = f.as<frameset>()))
```

Comment:
I thought you said a frameset can include other framesets?

Reply:
It's not supposed to be because on allocate_composite_frame() we 'open' each one of the frames and concatenate all of them together again.